### PR TITLE
auth: add email confirmation page

### PR DESCRIPTION
**Pull request by mvidalgarcia**
closes reanahub/reana#472

TODO:
- [x] Add confirm page with token as URL param.
  - if token invalid, display error notification from server and redirect to home.
  - if token valid, trigger success notification and redirect to home.

**Steps to test:**
1. Enable `notifications` in [Helm `values.yaml`](https://github.com/reanahub/reana/blob/2ff64da2563fe5c5a968b56ea90e983439deb22a/helm/reana/values.yaml#L83):
```yaml
notifications:
  enabled: true
  email_config:
    sender: admin@reana.io
```
2. Sign up a new local user (you should see a success notification)
3. Check [MailDev](http://localhost:32580), you should have received a confirmation email.
4. Copy the link in the email and replace the hostname and HTTPS->HTTP suitably
5. Paste it in your favorite browser, you should be redirected to the home page and a notification must show up, error or success, depending on the validity of the confirm token.
6. Sign in if the email was successfully confirmed.

### CODE REVIEW

**Review by audrium — COMMENTED** [review]
I can say that everything works nicely! Good job :rocket: 
 Just couple of minor questions: 

- After confirming the email and seeing the notification in the top of sign in form, once user logs in the notification bar is still shown. We probably want to clean the notifications once user logs in / logs out. This is not caused by PR, so we can create an separate issue for that.

- I noticed that it's possible to confirm other users account while being logged in, not sure if this is expected behaviour?
- Also it's possible to confirm already confirmed account, we should probably display some kind of different message in this case?

File: reana-ui/src/actions.js [hunk 1]
```diff
@@ -198,6 +202,37 @@ export function requestToken() {
...
+          triggerNotification(
+            "Success!",
+            "User email confirmed. You can sign in now."
+          )
+        );
+        return resp;
+      })
+      .catch((err) => {
+        // adapt error format (remove array)
+        err.response.data.message = err.response.data.message[0];
```
[the remark this answers is not in the document]
Comment (mvidalgarcia) [reply reana-ui/src/actions.js:229]: Unfortunately, this is the format that comes from the Invenio side and I'm not sure whether we can control it, that's why it differs from our typical format.
Adding some `.?` and check whether it's an array can definitely help, thanks!

**Comment by mvidalgarcia** [comment]
>     * After confirming the email and seeing the notification in the top of sign in form, once user logs in the notification bar is still shown. We probably want to clean the notifications once user logs in / logs out. This is not caused by PR, so we can create an separate issue for that.

I'll check if I can clean notifications before sign in. Thanks.

>     * I noticed that it's possible to confirm other users account while being logged in, not sure if this is expected behaviour?

Interesting, I had never tested that... I see that the confirmation happens, but later an "Invalid token" error is received and the user gets signed out. I'm afraid this is the Invenio behavior (`/confirm-email`) and there's no much we can do. In principle, this confirm token should only be accessed by the user confirming their email. Not sure if we should open a ticket to keep track of this, although it looks like a very rare corner case to me.

>     * Also it's possible to confirm already confirmed account, we should probably display some kind of different message in this case?

Changed, pushing soon, it was hardcoded, now we display the message coming from the server, which is something like "You've already confirmed this account" in that particular case.

**Review by audrium — APPROVED** [review]
Nice and clean fixes! The code looks even better now :rocket:

**Review by audrium — APPROVED** [review]
Just retested it and it all looks to be working just fine :+1:

**Pull request merged**